test/tactic_set.lean

```diff
@@ -0,0 +1,389 @@
+
+lemma sep_univ {α} {p : α → Prop} : {a ∈ (univ : set α) | p a} = {a | p a} := by set_taut
+
+
+lemma compl_set_of {α} (p : α → Prop) : {a | p a}ᶜ = { a | ¬ p a } := by set_taut
+
+theorem mem_compl_eq (s : set α) (x : α) : x ∈ sᶜ = (x ∉ s) := by set_taut
+
+theorem inter_compl_self (s : set α) : s ∩ sᶜ = ∅ := by set_taut
+
+theorem compl_inter_self (s : set α) : sᶜ ∩ s = ∅ := by set_taut
+
+theorem compl_empty : (∅ : set α)ᶜ = univ := by set_taut
+
+theorem compl_union (s t : set α) : (s ∪ t)ᶜ = sᶜ ∩ tᶜ := by set_taut'
+
+theorem compl_inter (s t : set α) : (s ∩ t)ᶜ = sᶜ ∪ tᶜ := by set_taut'
+
+theorem compl_univ : (univ : set α)ᶜ = ∅ := by set_taut'
+
+
+lemma compl_singleton_eq (a : α) : ({a} : set α)ᶜ = {x | x ≠ a} := by set_taut
+
+lemma compl_ne_eq_singleton (a : α) : ({x | x ≠ a} : set α)ᶜ = {a} :=
+by set_taut
+
+theorem union_eq_compl_compl_inter_compl (s t : set α) : s ∪ t = (sᶜ ∩ tᶜ)ᶜ :=
+by set_taut
+
+theorem inter_eq_compl_compl_union_compl (s t : set α) : s ∩ t = (sᶜ ∪ tᶜ)ᶜ :=
+by set_taut
+
+theorem union_compl_self (s : set α) : s ∪ sᶜ = univ := by set_taut
+
+theorem compl_union_self (s : set α) : sᶜ ∪ s = univ := by set_taut
+
+theorem compl_comp_compl : compl ∘ compl = @id (set α) := by set_taut
+
+theorem diff_eq (s t : set α) : s \ t = s ∩ tᶜ := by set_taut
+
+theorem diff_eq_compl_inter {s t : set α} : s \ t = tᶜ ∩ s := by set_taut
+
+theorem union_diff_cancel' {s t u : set α} (h₁ : s ⊆ t) (h₂ : t ⊆ u) : t ∪ (u \ s) = u := by set_taut
+
+
+theorem union_diff_cancel {s t : set α} (h : s ⊆ t) : s ∪ (t \ s) = t :=
+by set_taut
+
+
+theorem union_diff_left {s t : set α} : (s ∪ t) \ s = t \ s :=
+by set_taut
+
+theorem union_diff_right {s t : set α} : (s ∪ t) \ t = s \ t :=
+by set_taut
+
+theorem union_diff_distrib {s t u : set α} : (s ∪ t) \ u = s \ u ∪ t \ u :=
+by set_taut'
+
+theorem inter_union_distrib_left {s t u : set α} : s ∩ (t ∪ u) = (s ∩ t) ∪ (s ∩ u) := by set_taut'
+
+theorem inter_union_distrib_right {s t u : set α} : (s ∩ t) ∪ u = (s ∪ u) ∩ (t ∪ u) := by set_taut'
+
+theorem union_inter_distrib_left {s t u : set α} : s ∪ (t ∩ u) = (s ∪ t) ∩ (s ∪ u) :=  by set_taut'
+
+theorem union_inter_distrib_right {s t u : set α} : (s ∪ t) ∩ u = (s ∩ u) ∪ (t ∩ u) := by set_taut'
+
+theorem inter_diff_assoc (a b c : set α) : (a ∩ b) \ c = a ∩ (b \ c) :=
+by set_taut
+
+theorem inter_diff_self (a b : set α) : a ∩ (b \ a) = ∅ :=
+by set_taut
+
+theorem inter_union_diff (s t : set α) : (s ∩ t) ∪ (s \ t) = s := by set_taut
+
+theorem inter_union_compl (s t : set α) : (s ∩ t) ∪ (s ∩ tᶜ) = s := by set_taut
+
+
+
+theorem compl_eq_univ_diff (s : set α) : sᶜ = univ \ s :=
+by set_taut
+
+lemma empty_diff (s : set α) : (∅ \ s : set α) = ∅ := by set_taut
+
+
+theorem diff_empty {s : set α} : s \ ∅ = s := by set_taut
+
+theorem diff_diff {u : set α} : s \ t \ u = s \ (t ∪ u) := by set_taut
+
+-- the following statement contains parentheses to help the reader
```

Comment:
I'm personally a fan of adding extra parentheses when it would help the reader, so thanks!

Reply:
I copy/pasted from set/basic.lean, so I believe another deserves your thanks. :-)